### PR TITLE
Support "warName" property of maven-war-plugin

### LINKED ISSUE #2206: jib-maven-plugin fails if maven-war-plugin "warName" property is set
**Environment**:

- *Jib version:* 1.8.0, 1.7.0
- *Build tool:*  Maven 3.6.3
- *OS:* Fedora / Windows

**Description of the issue**:
Between 1.6.1 and 1.7.0, the behaviour of the jib-maven-plugin changed and no longer works if the WAR name is not according to the default (artifactID-version.war), but changed using the "warName" property of the maven-war-plugin.

**Expected behavior**:
Upgrading the jib-plugin from 1.6.1 to 1.7.0 does not change the result of the build.

**Steps to reproduce**:
I created a small repository that you can clone: https://github.com/padyx/jib-test-repository

Working configuration (initial repository state):

1. Alter the pom.xml File in _helloworld.ui.html.app.war_ and change the target image to a location you can push to. 
2. Execute the build as follows: `mvn -Djib.to.auth.username=xxxxx -Djib.to.auth.password=xxxxx -DskipTests -P docker clean package`  (Success)

Error if upgrading:
1. Change the jib-maven-plugin Version to 1.7.0 (or 1.8.0) in the helloworld/pom.xml File.
2. Rerun the build with the command above.

**`jib-maven-plugin` Configuration**: 
```xml
<!-- From parent module (build / pluginManagement) -->
<plugin>
  <groupId>com.google.cloud.tools</groupId>
    <artifactId>jib-maven-plugin</artifactId>
    <version>1.7.0</version>
    <executions>
      <execution>
      <phase>package</phase>
      <goals>
        <goal>build</goal>
      </goals>
    </execution>
  </executions>
</plugin>

<!-- From build in actual module-->
<plugin>
  <groupId>com.google.cloud.tools</groupId>
  <artifactId>jib-maven-plugin</artifactId>
  <configuration>
    <from>
      <image>tomcat:9-jdk11</image>
    </from>
    <to>
      <image>padyx/testimage</image>
    </to>
  </configuration>
</plugin>
```

**Log output**: 
`[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.7.0:build (default) on project helloworld.ui.html.app.war: Obtaining project build output files failed; make sure you have compiled your project before trying to build the image. (Did you accidentally run "mvn clean jib:build" instead of "mvn clean compile jib:build"?): /home/chakotay/eclipse-workspace/helloworld/helloworld.ui.html.app.war/target/org.eclipse.scout.apps.helloworld.ui.html.war -> [Help 1]`

**Additional information**
The configuration for the war name is: (helloworld.ui.html.app.war POM File)
```xml
<plugin>
  <artifactId>maven-war-plugin</artifactId>
  <configuration>
    <warName>uihtml</warName>
  </configuration>
</plugin>
```

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #2206.

### CODE REVIEW

File: jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java [hunk 1]
```diff
@@ -419,6 +418,24 @@ public boolean isOffline() {
     return session.isOffline();
   }
 
+  @VisibleForTesting
+  Path getWarArtifact() {
+    Build build = project.getBuild();
+    String warName = build.getFinalName();
+
+    Plugin warPlugin = project.getPlugin("org.apache.maven.plugins:maven-war-plugin");
```
Comment (loosebazooka) [inline jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java]: this wont work if they set the property on the command line? or does it?

Comment (chanseokoh) [reply jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java]: Is it possible to set the configuration property on the command line? Does this work for any Maven plugin?

Comment (loosebazooka) [reply jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java]: I'm not sure, we may not HAVE to support it, but it'd be useful to see if it's possible and document it.

Comment (chanseokoh) [reply jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java]: From my investigation, `warName` does not have the user property (such as `maven.compiler.release`) in recent versions. Actually, `warName` isn't documented in the [latest `maven-war-plugin` doc](https://maven.apache.org/plugins/maven-war-plugin/war-mojo.html). Looks like it was in an [old version](https://maven.apache.org/plugins-archives/maven-war-plugin-2.4/war-mojo.html#warName), but I think now it's retained only for backward compatibility in recent versions. The doc for the old version does say the user property is `war.warName`, but setting it in a recent version does nothing.

Comment (loosebazooka) [reply jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java]: sgtm

**Pull request merged**